sky/packages/sky/lib/src/material/bottom_sheet.dart

```diff
@@ -0,0 +1,124 @@
+// Copyright 2015 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:math' as math;
+
+import 'package:flutter/animation.dart';
+import 'package:flutter/painting.dart';
+import 'package:flutter/rendering.dart';
+import 'package:flutter/widgets.dart';
+
+import 'colors.dart';
+import 'theme.dart';
+
+const Duration _kBottomSheetDuration = const Duration(milliseconds: 200);
+
+class _BottomSheet extends StatefulComponent {
+  _BottomSheet({
+    Key key,
+    this.child,
+    this.performance,
+    this.navigatorSize
+  }) : super(key: key);
+
+  final Widget child;
+  final PerformanceView performance;
+  final Size navigatorSize;
+
+  _BottomSheetState createState() => new _BottomSheetState();
+}
+
+class _BottomSheetState extends State<_BottomSheet> {
+
+  double _endHeight;
+
+  void _handleChildSizeChanged(Size newSize) {
+    final double maxSheetHeight = config.navigatorSize.height * 9.0 / 16.0;
+    setState(() {
+      _endHeight = math.min(maxSheetHeight, newSize.height);
+    });
+  }
+
+  Widget build(BuildContext context) {
+     return new SquashTransition(
+        performance: config.performance,
+        height: new AnimatedValue<double>(0.0, end: _endHeight ?? 0.0),
+        child: new ScrollableViewport(
+          child: new Container(
+            decoration: new BoxDecoration(backgroundColor: Theme.of(context).canvasColor),
```

Comment:
I suppose so.  At the moment, using Material here causes the first SquashTransition to flash (i.e. there's no animation). After the first showModalBottomSheet() it DTRT.